dotnet/example_code/ec2/Ec2SpotCRUD.cs

```diff
@@ -0,0 +1,220 @@
+            // to monitor (e.g. that we started).
+            var request = new DescribeSpotInstanceRequestsRequest();
+            request.SpotInstanceRequestIds.Add(spotRequestId);
+
+            // Retrieve the request we want to monitor.
+            var describeResponse = ec2Client.DescribeSpotInstanceRequests(request);
+
+            SpotInstanceRequest req = describeResponse.SpotInstanceRequests[0];
+
+            return req.State;
+        }
+        // snippet-end:[ec2.dotnet.spot_instance_get_spot_request_state]
+
+        // snippet-start:[ec2.dotnet.spot_instance_cancel_spot_request]
+        public static void CancelSpotRequest(
+            AmazonEC2Client ec2Client,
+            string spotRequestId)
+        {
+            var cancelRequest = new CancelSpotInstanceRequestsRequest();
+
+            cancelRequest.SpotInstanceRequestIds.Add(spotRequestId);
+
+            ec2Client.CancelSpotInstanceRequests(cancelRequest);
+        }
+        // snippet-end:[ec2.dotnet.spot_instance_cancel_spot_request]
+
+        // snippet-start:[ec2.dotnet.spot_instance_terminate_spot_request]
+        public static void TerminateSpotInstance(
+            AmazonEC2Client ec2Client,
+            string spotRequestId)
+        {
+            var describeRequest = new DescribeSpotInstanceRequestsRequest();
+            describeRequest.SpotInstanceRequestIds.Add(spotRequestId);
+
+            // Retrieve the request we want to monitor.
+            var describeResponse = ec2Client.DescribeSpotInstanceRequests(describeRequest);
+
+            if (SpotInstanceState.Active == describeResponse.SpotInstanceRequests[0].State)
+            {
+                string instanceId = describeResponse.SpotInstanceRequests[0].InstanceId;
+
+                var terminateRequest = new TerminateInstancesRequest();
+                terminateRequest.InstanceIds = new List<string>() { instanceId };
+
+                try
+                {
+                    var terminateResponse = ec2Client.TerminateInstances(terminateRequest);
+                }
+                catch (AmazonEC2Exception ex)
+                {
+                    // Check the ErrorCode to see if the instance does not exist.
+                    if ("InvalidInstanceID.NotFound" == ex.ErrorCode)
+                    {
+                        Console.WriteLine("Instance {0} does not exist.", instanceId);
+                    }
+                    else
+                    {
+                        // The exception was thrown for another reason, so re-throw the exception.
+                        throw;
+                    }
+                }
+            }
+        }
+        // snippet-end:[ec2.dotnet.spot_instance_terminate_spot_request]
+
+        // snippet-start:[ec2.dotnet.spot_instance_main]
+        // This program takes one argument, the AMI
+        // Something like: ami-xxxxxxxx
+        static void Main(string[] args)
```

Comment:
Nice job on providing a unit test here. xUnit.NET, NUnit, or MSTest would be really cool here instead of Main, but this will certainly do. Also, it might be nice to make this just a tiny bit more reusable by showing how to call Main with input args instead of hard-coding values inside of Main, but again this will certainly do.

Reply:
I've added as many command-line args as possible.